core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/dataset/DatasetsControllerTest.java

```diff
@@ -186,6 +182,35 @@ void testGetByIdNotFound() {
                 .isInstanceOf(MetadataDoesNotExistException.class);
     }
 
+    @Test
+    @WithIesiUser(username = "spring",
+            authorities = {"DATASETS_READ@PUBLIC"})
+    void testGetByUuidImplementations() {
+        UUID uuid = UUID.randomUUID();
+
+        List<DatasetImplementationDto> datasetImplementationDtoList = new ArrayList<>();
+        datasetImplementationDtoList.add(new InMemoryDatasetImplementationDto());
+
+        when(datasetDtoService.fetchImplementationsByUuid(uuid))
+                .thenReturn(datasetImplementationDtoList);
+
+        assertThat(datasetController.getImplementationsByUuid(uuid))
+                .containsAll(datasetImplementationDtoList);
+    }
+
+    @Test
+    @WithIesiUser(username = "spring",
+            authorities = {"DATASETS_READ@PUBLIC"})
+    void testGetByUuidImplementationsNotFound() {
+        UUID uuid = UUID.randomUUID();
+
+        when(datasetDtoService.fetchImplementationsByUuid(uuid))
+                .thenReturn(null);
+
+        assertThatThrownBy(() -> datasetController.get(uuid))
+                .isInstanceOf(MetadataDoesNotExistException.class);
+    }
+
```

Comment:
Also create security tests in io.metadew.iesi.server.rest.dataset.DatasetsControllerSecurityTest